tipg/middleware.py

```diff
@@ -41,3 +46,24 @@ async def dispatch(self, request: Request, call_next):
                 response.headers["Cache-Control"] = self.cachecontrol
 
         return response
+
+
+class CatalogUpdateMiddleware(BaseHTTPMiddleware):
+    """Middleware to update the catalog cache."""
+
+    async def dispatch(self, request: Request, call_next):
+        """Fetch Catalog either immediately or in background."""
+        if request.query_params.get("refresh"):
+            logger.debug("Forcing catalog refresh.")
+            await register_collection_catalog(request.app)
+            response = await call_next(request)
```

Comment:
I don't think we need to set a special case for this. IMO the `refresh` will update the `catalog_expire` so it should not reach the next if block

Reply:
The /refresh endpoint is only activated if settings.debug is TRUE. 

With this middleware there is no way to no for sure that you are getting a non-cached catalog without having this option since normally, the refresh is only ever happening in a background task and there is no way to know that it has run, while with this option it will block processing the request until it has refreshed the catalog if you want to make sure that you have the latest catalog.

Reply:
The issue with passing a `refresh` query parameter is that then you have to make sure any of the application endpoint won't need or refuse such keyword 🤷 (e.g: https://github.com/developmentseed/tipg/blob/main/tipg/dependencies.py#L223-L249)

Reply:
also imagine if someone pass `?refresh=False` it will refresh the catalog because the `False` will be passed as string not as bool

Reply:
> With this middleware there is no way to know for sure that you are getting a non-cached catalog

IMO, that's not a big deal! we could add a `last_updated` info in the `/collections` endpoint if this make things easier

Reply:
All fair points. We can get rid of it. I might still want to add that functionality back in later -- as I think about it, it may be better to check the headers for Cache-Control=no-cache to trigger something like this.

Reply:
a custom `X-header` would be better. 

I'm about to submit a PR that change some of the logic.